File: WEEK_0/SortingRoutine/JavaScript/sort.js
function bubbleSort(arr, n){
    for(let i = 0; i < n - 1; i++){
        for(let j = 0; j < n - i - 1; j++){
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    return arr;
}

function insertSort(arr, n){
    for(let i = 1; i < n; i++){
        let key = arr[i];
        let j = i - 1;
        while (j >= 0 && arr[j] > key){
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
    return arr;
}

let array = [5, 3, 8, 4, 2];
n = array.length;

console.log("Original array:");
console.log(array);

console.log("Sorted array:");
console.log(bubbleSort(array, n));

File: WEEK_0/SortingRoutine/Rust/sort.rs
fn bubble_sort(slice: &mut [i32]){
    let size_arr = slice.len();
    for i in 0..(size_arr-1) {
        for j in 0..(size_arr - i - 1) {
            if slice[j] > slice[j + 1] {
                slice.swap(j, j + 1);
            }
        }
    }
}


fn main(){
    let mut arr: [i32; 7] = [5, 2, 9, 1, 5, 6, 3];
    println!("Original array: {:?}", arr);
    bubble_sort(&mut arr);
    println!("Sorted array: {:?}", arr);
}
